Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,433 @@
+        return node 
+
+    def visit_CEnumDefNode(self, node):
+        # TODO Figure out how to define an enum-class via typehints...
+
+        # NOTE It seems that only public will make the enum accessible to python so 
+        # I'll just have it check if the enums will be public for now... - Vizonex
+        if node.visibility == "public":
+            # Enum's name is not in or visible in the final product because 
+            # it's not an enum class so do not indent here...
+            # Also Leave visit_CEnumDefItemNode up to the previous 
+            # class's function...
+            self.putline("# -- enum %s --" % node.name)
+            self.visitchildren(node)
+        return node 
+
+    # Used in our translation table to register return types variables from...
+    def visit_CTypeDefNode(self,node):
+        if isinstance(node.declarator, CNameDeclaratorNode):
+            # Register a new type to use in our translation table...
+            self.translation_table[node.declarator.name] = self.translate_base_type_to_py(node.base_type)
+    
+    def visit_CStructOrUnionDefNode(self, node):
+        # XXX : Currently, I don't know what to do here yet but ignoring 
+        # is triggering some problems currently...
+        return node
+        
+
+    def visit_CVarDefNode(self, node):
+
+        # if they aren't public or readonly then the variable inside of a class 
+        # or outside should be ignored by default...
+
+        if node.visibility in ["readonly", "public"]:
+
+            # TODO handle ctypedef nodes and give them a 
+            # new type-registry system to help translate 
+            # all incoming variables... 
+
+            py_name = self.translate_base_type_to_py(node.base_type)
+            
+            # Final check...
+            if py_name is not None:
+                # Write in all the objects listed on the defined line...
+                for d in node.declarators:
+                    self.putline("%s: %s" % (d.name, py_name))
+    
+        return node
+
+
+    
+
+
+    def visit_ImportNode(self, node):
+        module_name = node.module_name.value
+
+        if not node.name_list:
+            self.putline("import %s" % module_name) 
+        else:
+            all_imported_children = ", ".join((arg.value for arg in node.name_list.args))
+
+            if node.level > 0:
+                module_name = "%s%s" % ("." * node.level , module_name)
+
+            self.putline("from %s import %s" % (module_name, all_imported_children))
+
+        return node
+
+    # Optimized original code by having there be one function to take 
+    # the place of two of them I could see what Scoder meant when 
+    # said the original pull request needed to be cleaned up...
+
+    
+    def write_class(self, node, class_name):
+        self.endline()
+        self.put("class %s" % class_name)
+        if getattr(node,"bases",None) and isinstance(node.bases, TupleNode) and node.bases.args:
```

Comment:
I don't think `node.bases` needs a `getattr` - it's a child of both `PyClassDefNode` and `CClassDefNode`